compiler/optimizer/DeadTreesElimination.cpp

```diff
@@ -264,9 +264,9 @@ static bool isSafeToReplaceNode(TR::Node *currentNode, TR::TreeTop *curTreeTop,
    if (isUnresolvedReference)
       return false;
 
-   bool mayBeVolatileReference = currentNode->mightHaveVolatileSymbolReference();
-   // Do not swing down volatile nodes
-   if (mayBeVolatileReference)
+   bool mayBeNonTransparentReference = currentNode->mightHaveNonTransparentSymbolReference();
+   // Do not swing down non-transparent nodes
+   if (mayBeNonTransparentReference)
```

Comment:
I don't think this needs to be conservative in the treatment of `OpaqueSemantics`.  I think it really only needs to worry about situations where there could be interactions with other threads, so `AcquireReleaseSemantics` or stronger.

I suspect the same might hold true for many of the changes for other optimizations, so I think they need to be considered on a case-by-case basis.  I won't go through and comment on each.